### PR TITLE
Fix pathname* to throw a type-error, when nil is passed as stream, instead of simple-error

### PR DESCRIPTION
**Pull request by kpoeck**
e.g.
```lisp
COMMON-LISP-USER> (pathname nil)

Condition of type: SIMPLE-TYPE-ERROR
In function PATHNAME,the value of the only argument is
  NIL
which is not of the expected type (OR FILE-STREAM STRING PATHNAME SYNONYM-STREAM)

Available restarts:
(use :r1 to invoke restart 1)

1. (RESTART-TOPLEVEL) Go back to Top-Level REPL.
````
validated running the ansi-tests

### CODE REVIEW

File: src/core/pathname.cc [hunk 1]
```diff
@@ -988,11 +989,12 @@ CL_DEFUN Pathname_sp cl__pathname(T_sp x) {
     x = cl__parse_namestring(x);
   } else if (gc::IsA<Pathname_sp>(x)) {
     // do nothing
-  } else if (gc::IsA<Stream_sp>(x)) {
+  } else if ((gc::IsA<FileStream_sp>(x)) or (gc::IsA<SynonymStream_sp>(x))){
```
Comment (drmeister) [inline src/core/pathname.cc]: I think this will be the first 'or' in the Clasp code. 
Should we allow this?  Or stick with || ???

**Comment by kpoeck** [comment]
Will change (I am happy to follow any code conventions as long as I know them)

**Comment by kpoeck** [comment]
@drmeiter Now have       ((gc::IsA<FileStream_sp>(x)) || (gc::IsA<SynonymStream_sp>(x))), please check

**Comment by drmeister** [comment]
Thanks!

**Pull request merged**